backends/magma/ceed-magma.c

```diff
@@ -865,86 +869,52 @@ static int CeedOperatorApply_Magma(CeedOperator op, CeedVector invec,
       // Zero evec
-      ierr = CeedVectorGetArray(opmagma->evecs[iein], CEED_MEM_HOST, &vec_temp);
+      ierr = CeedVectorGetArray(opmagma->evecs[i], CEED_MEM_HOST, &vec_temp);
       CeedChk(ierr);
-      for (CeedInt j=0; j<opmagma->evecs[iein]->length; j++)
+      for (CeedInt j=0; j<opmagma->evecs[i]->length; j++)
         vec_temp[j] = 0.;
-      ierr = CeedVectorRestoreArray(opmagma->evecs[iein], &vec_temp); CeedChk(ierr);
+      ierr = CeedVectorRestoreArray(opmagma->evecs[i], &vec_temp); CeedChk(ierr);
       // Active
       if (op->inputfields[i].vec == CEED_VECTOR_ACTIVE) {
         // Restrict
         ierr = CeedElemRestrictionApply(op->inputfields[i].Erestrict, CEED_NOTRANSPOSE,
-                                        lmode, invec, opmagma->evecs[iein],
+                                        lmode, invec, opmagma->evecs[ieiin],
                                         request); CeedChk(ierr);
         // Get evec
-        ierr = CeedVectorGetArrayRead(opmagma->evecs[iein], CEED_MEM_HOST,
+        ierr = CeedVectorGetArrayRead(opmagma->evecs[i], CEED_MEM_HOST,
                                       (const CeedScalar **) &opmagma->edata[i]); CeedChk(ierr);
-        iein++;
       } else {
         // Passive
         // Restrict
         ierr = CeedElemRestrictionApply(op->inputfields[i].Erestrict, CEED_NOTRANSPOSE,
-                                        lmode, op->inputfields[i].vec, opmagma->evecs[iein],
+                                        lmode, op->inputfields[i].vec, opmagma->evecs[i],
                                         request); CeedChk(ierr);
         // Get evec
-        ierr = CeedVectorGetArrayRead(opmagma->evecs[iein], CEED_MEM_HOST,
+        ierr = CeedVectorGetArrayRead(opmagma->evecs[i], CEED_MEM_HOST,
                                       (const CeedScalar **) &opmagma->edata[i]); CeedChk(ierr);
-        iein++;
       }
     }
   }
 
   // Output Evecs
-  for (CeedInt i=0,ieout=opmagma->numein; i<qf->numoutputfields; i++) {
-    // No Restriction
-    if (op->outputfields[i].Erestrict == CEED_RESTRICTION_IDENTITY) {
-      // Active
-      if (op->outputfields[i].vec == CEED_VECTOR_ACTIVE) {
-        ierr = CeedVectorGetArray(outvec, CEED_MEM_HOST,
-                                  &opmagma->edata[i + qf->numinputfields]); CeedChk(ierr);
-      } else {
-        // Passive
-        ierr = CeedVectorGetArray(op->outputfields[i].vec, CEED_MEM_HOST,
-                                  &opmagma->edata[i + qf->numinputfields]); CeedChk(ierr);
-      }
-    } else {
-      // Restriction
-      ierr = CeedVectorGetArray(opmagma->evecs[ieout], CEED_MEM_HOST,
-                                &opmagma->edata[i + qf->numinputfields]); CeedChk(ierr);
-      ieout++;
+  for (CeedInt i=0; i<qf->numoutputfields; i++) {
+    ierr = CeedVectorGetArray(opmagma->evecs[i+opmagma->numein], CEED_MEM_HOST,
+                              &opmagma->edata[i + qf->numinputfields]);
+    CeedChk(ierr);
     }
```

Comment:
Please check that the MAGMA backend actually compiles (even if you don't run it). Looks like this is a dangling brace.